### PR TITLE
Fix pointer sheet tracking and synthesis in SYNTHESIZE-BOUNDARY-EVENTS

### PR DESCRIPTION
**Pull request by scymtym**
* Do not distribute inappropriate X enter/leave events

* Handle grabbing-related events appropriately

* Perform port pointer sheet tracking and boundary event synthesis separately

### CODE REVIEW

File: Core/clim-basic/events.lisp [hunk 1]
```diff
@@ -188,8 +190,11 @@
 (define-event-class pointer-enter-event    (pointer-boundary-event) ())
 (define-event-class pointer-exit-event     (pointer-boundary-event) ())
 
-(define-event-class pointer-ungrab-enter-event (pointer-enter-event) ())
-(define-event-class pointer-ungrab-leave-event (pointer-exit-event)  ())
+(defclass grab-event () ())
+(define-event-class pointer-grab-enter-event (pointer-enter-event grab-event) ())
+(define-event-class pointer-grab-leave-event (pointer-exit-event grab-event) ())
+(define-event-class pointer-ungrab-enter-event (pointer-enter-event grab-event) ())
+(define-event-class pointer-ungrab-leave-event (pointer-exit-event grab-event) ())
```
Comment (dkochmanski) [inline Core/clim-basic/events.lisp]: is there rationale behind adding grab-event class?
please add a comment describing what these evens mean.

**Comment by dkochmanski** [comment]
n.b I couldn't trigger any regression during exploratory testing.

File: Core/clim-basic/events.lisp (hunk 1 above)
Comment (scymtym) [reply Core/clim-basic/events.lisp]: It used to be used for discrimination but not anymore. Not sure whether it is still useful. I can just remove it, if you prefer.

**Comment by dkochmanski** [comment]
Please do

On September 28, 2019 2:16:47 PM GMT+02:00, Jan Moringen <notifications@github.com> wrote:
>scymtym commented on this pull request.
>
>
>
>> @@ -188,8 +190,11 @@
>(define-event-class pointer-enter-event    (pointer-boundary-event) ())
>(define-event-class pointer-exit-event     (pointer-boundary-event) ())
> 
>-(define-event-class pointer-ungrab-enter-event (pointer-enter-event)
>())
>-(define-event-class pointer-ungrab-leave-event (pointer-exit-event) 
>())
>+(defclass grab-event () ())
>+(define-event-class pointer-grab-enter-event (pointer-enter-event
>grab-event) ())
>+(define-event-class pointer-grab-leave-event (pointer-exit-event
>grab-event) ())
>+(define-event-class pointer-ungrab-enter-event (pointer-enter-event
>grab-event) ())
>+(define-event-class pointer-ungrab-leave-event (pointer-exit-event
>grab-event) ())
>
>It used to be used for discrimination but not anymore. Not sure whether
>it is still useful. I can just remove it, if you prefer.

-- Wysłane za pomocą K-9 Mail.

File: Core/clim-basic/events.lisp (hunk 1 above)
Comment (scymtym) [reply Core/clim-basic/events.lisp]: Removed.

**Comment by dkochmanski** [comment]
Thanks!

**Pull request merged**